### DISCUSSION TITLE
new popup overflows terminal width

### DISCUSSION THREAD

**Discussion by damnskippy** [neoclide/coc.nvim#3958] [Q&A] [answered]
Using the new popup implementation off the latest commit (as of Aug 1st) from master branch.
If terminal window running vim is not wide enough, popup menu width overflows and exceeds the width of the pane.  If terminal window is maximized then entire popup menu is visible.  This issue was not present with the old pmenu.
vim's pumwidth option only sets the *min* width, unlike pumheight which sets the *max* width.
Not sure if anyone else sees this behavior.
Thanks.

Edit: updated to remove tmux references as it's not accurate and add screeshots.

Screencap below to show the overflow - see on the write icons and are cut off.
Vim has &columns value of 91 in this case.

![image](https://user-images.githubusercontent.com/5184519/182415502-0c36874c-8bc6-4edd-a50d-3b90b5a7cada.png)

When window is wide enough (&columns = 198):

![image](https://user-images.githubusercontent.com/5184519/182415891-fefd7b12-e52d-47a8-8e8c-20a033ab628a.png)

**Comment by chemzqm** [comment, marked as answer]
You need to configure `"suggest.labelMaxLength"` on that case
